notifications.yaml

```diff
@@ -0,0 +1,268 @@
+swagger: '2.0'
+info:
+  version: 0.0.1
+  title: Notification API
+  description: Notification API.
+host: localhost
+basePath: /api/v1
+schemes:
+  - https
+paths:
+  '/messages/{id}':
+    parameters:
+      - name: id
+        in: path
+        type: string
+        required: true
+    get:
+      operationId: getMessage
+      summary: Get Message
+      description: Returns a previously submitted message.
+      responses:
+        '200':
+          description: Message found.
+          schema:
+            $ref: '#/definitions/AcceptedMessage'
+        '404':
+          description: No message found.
+  '/users/{fiscal_number}/messages':
+    parameters:
+      - name: fiscal_number
+        in: path
+        type: string
+        required: true
+        description: The fiscal number of the user.
+        pattern: >-
+          [A-Z]{6}[0-9LMNPQRSTUV]{2}[ABCDEHLMPRST][0-9LMNPQRSTUV]{2}[A-Z][0-9LMNPQRSTUV]{3}[A-Z]
+        maxLength: 16
+        minLength: 1
+      - $ref: '#/parameters/PaginationRequest'
+    get:
+      operationId: getMessagesByUser
+      summary: Get messages by user
+      description: |-
+        Returns the submitted messages the have the user as destination.
+        Messages will be returned in inverse delivery order (from last to first)
+      responses:
+        '200':
+          description: Found.
+          schema:
+            allOf:
+              - type: object
+                properties:
+                  items:
+                    type: array
+                    items:
+                      $ref: '#/definitions/AcceptedMessage'
+              - $ref: '#/definitions/PaginationResponse'
+          examples: {}
+        '404':
+          description: No user found.
+      parameters: []
+  /messages:
+    post:
+      operationId: submitMessage
+      summary: Submit Message
+      description: Submits a message for delivery to a user.
+      parameters:
+        - name: message
+          in: body
+          schema:
+            $ref: '#/definitions/MessageDefinition'
+      responses:
+        '201':
+          description: Message accepted for delivery.
+          schema:
+            allOf:
+              - properties:
+                  id:
+                    type: string
+                    format: uuid
+                  status:
+                    $ref: '#/definitions/MessageStatus'
+                required:
+                  - id
+                  - status
+            description: ''
+            type: object
+          headers:
+            Location:
+              type: string
+              description: >-
+                Location (URL) of accepted message resource.
```

Comment:
I think "accepted" is more correct here as submitted messages can be rejected. Moreover, if we call the input model 'SubmittedMessage' and the output model 'AcceptedMessage' this rule is enforced.